share/functions/fish_vi_key_bindings.fish

```diff
@@ -6,6 +6,127 @@ function fish_vi_inc
     fish_vi_inc_dec inc
 end
 
+function fish_vi_arg_digit --description 'Accumulate a digit for the next command'
+    set -g __fish_vi_count "$__fish_vi_count$argv[1]"
+end
+
+function fish_vi_run_count --description 'Run a command $__fish_vi_count times'
+    set -l count $__fish_vi_count
+    if test -z "$count"
+        set count 1
+    end
+    # Reset count for next time
+    set -g __fish_vi_count
+
+    for i in (seq $count)
+        # Check if the first argument is a defined shell function (like fish_vi_move_end_word)
+        if functions -q -- $argv[1]
+            # Execute the shell function directly
+            $argv
+        else
+            # Execute it as an internal bind command (like backward-char)
+            commandline -f $argv
+        end
+    end
+end
+
+function fish_vi_start_operator
+    set -g __fish_vi_operator $argv[1]
+    set -g __fish_vi_start_count $__fish_vi_count
+    set -g __fish_vi_count
```

Comment:
maybe we should reuse it here as well

	set -g __fish_vi_start_count (__fish_vi_consume_count __fish_vi_count)

and in fish_vi_operator_cancel we could do "__fish_vi_consume_count >/dev/null".. not sure if that's better